PUBLISHING.md

```diff
@@ -0,0 +1,78 @@
+# Publishing new versions of uefi-rs to Crates.io
+
+This guide documents the best practices for publishing new versions of
+the crates in this repository to [crates.io](https://crates.io/).
+
+**It is mostly intended for maintainers of the uefi-rs project.**
+
+## Before bumping the crate versions
+
+It is a good idea to occasionally update/refresh the dependencies of the uefi-rs crates,
+even if we don't necessarily need a bugfix included in one of them.
+
+This can be done easily by using the [`cargo upgrade`][cargo-upgrade] tool.
+
+[cargo-upgrade]: https://crates.io/crates/cargo-edit#cargo-upgrade
+
+## Bumping the crate versions
+
+For ensuring compatibility within the crates ecosystem,
+Cargo [recommends][cargo-semver] maintainers to follow the [semantic versioning][semver] guidelines.
+
+This means that before publishing the changes, we need to decide
+which crates were modified and how should their version numbers be incremented.
+
+Incrementing the version number of a crate is as simple as editing
+the corresponding `Cargo.toml` file and updating the `version = ...` line,
+then commiting the change (preferrably on a new branch, so that all of the version bumps
+can be combined in a single pull request).
+
+### Crate dependencies
+
+The dependency graph of the published crates in this repo is:
+
+```
+uefi-macros <-- uefi (root project) <-- uefi-services
```

Comment:
I had a little trouble parsing this, might be clearer fully written out? E.g.
> uefi-services depends on uefi (the root project), and uefi depends on uefi-macros.

Reply:
Reworded this, should be clearer now.